develop/plugin_tutorial.rst

```diff
@@ -158,402 +104,358 @@ special note of the section that looks like this::
+
+            Returns a sample of the <number of items> taken from the remaining
+            arguments.  <number of items> must be less than the number
+            of arguments given.
+            """
+            if n > len(items):
+                # Calling irc.error with Raise=True is an alternative early return
+                irc.error('<number of items> must be less than the number '
+                          'of arguments.', Raise=True)
+            sample = self.rng.sample(items, n)
+            sample.sort()
+            irc.reply(utils.str.commaAndify(sample))
+
+The important thing to note is that list type arguments are rolled into one
+parameter in the command function by the ``many`` filter. Similar "multiplicity"
+handlers are documented :ref:`here <wrap-multiplicity-handlers>`.
+
+We also update the docstring to use the ``[]`` convention when surrounding
+optional arguments.
+
+For this function's body,
+:py:meth:`irc.error <supybot.callbacks.NestedCommandsIrcProxy.error>`
+is like
+:py:meth:`irc.replySuccess <supybot.callbacks.NestedCommandsIrcProxy.replySuccess>`
+but for error messages. We prefer using this instead of ``irc.reply`` for error
+signaling because its behaviour can be configured specially. For example, you
+can force all errors to go in private by setting the ``reply.error.inPrivate``
+option, and this can help reduce noise on a busy channel.
+Also, ``irc.error()`` with no text will return a generic error message
+configured in ``supybot.replies.error``, but this is not a valid call to
+:py:meth:`irc.reply <supybot.callbacks.NestedCommandsIrcProxy.reply>`.
+
+``utils.str.commaAndify`` is a simple helper that takes a list of strings
+and turns it into "item1, item2, item3, item4, and item5" for an arbitrary
+length. Limnoria has accumulated many such helpers in its lifetime, many of
+which are described in the :ref:`Using Utils <using-utils>` page.
+
+Command handler with optional arguments
+---------------------------------------
+Now for the last command that we will add to our plugin.py. This ``diceroll``
+command will allow the bot users to roll an arbitrary n-sided die, with n
+defaulting to 6:
+
+.. code-block::
+    :dedent: 0
+
+        @wrap([additional(('int', 'number of sides'), 6)])
+        def diceroll(self, irc, msg, args, n):
+            """[<number of sides>]
+
+            Rolls a die with <number of sides> sides.  The default number of sides
+            is 6.
+            """
+            s = 'rolls a %s' % self.rng.randrange(1, n)
+            irc.reply(s, action=True)
+
+The only new thing described here is that ``irc.reply(..., action=True)`` makes
+the bot perform a `/me`. There are some other flags described in the
+:py:meth:`irc.reply <supybot.callbacks.NestedCommandsIrcProxy.reply>`
+documentation too: common ones include ``private=True``, which
+forces a private message, and ``notice=True``, which forces the reply to use
+NOTICE instead of PRIVMSG.
```

Comment:
```suggestion
The only new thing described here is that ``irc.reply(..., action=True)`` makes
the bot perform a `/me`. There are some other flags described in the
:py:meth:`irc.reply <supybot.callbacks.ReplyIrcProxy.reply>`
documentation too: common ones include ``private=True``, which
forces a private message, and ``notice=True``, which forces the reply to use
NOTICE instead of PRIVMSG.
```